compiler/rustc_target/src/asm/mod.rs

```diff
@@ -596,6 +617,7 @@ impl InlineAsmType {
 
     pub fn size(self) -> Size {
         Size::from_bytes(match self {
+            Self::I1 => return Size::from_bits(1),
```

Comment:
In my asm! tests I cannot find a primitive type to use for the test. i1 does not exist and bool does not work.
Any ideas from your side?

Reply:
Rust doesn't have a primitive type of that size (bool is still a byte). However we can still test the feature, I think something like this should work (not tested atm):

```rust
// last 16 bits **must** be 0 else undefined behaviour can occur
let x: u32 = 0b11111111_11111111_00000000_0000000;
let mut y: u32 = 12;
let new: u32 = 5;
unsafe {
    asm!(
        "wsr.BR {0}",
        "movt {1}, {2}, b0"
        , in(reg) x, inout(reg) y, in(reg) new
    );
};
assert_eq!(y, new);
```